src/redux/reducers/dataReducer.js

```diff
@@ -47,6 +48,15 @@ export default (state = INITIAL_STATE, action) => {
       newState.blocks[action.payload.id] = action.payload;
       return newState;
 
+    case DATA_DELETE_BLOCK:
+      newState = {
+        ...state,
+      };
+
+      delete newState.blocks[action.payload.blockId];
+      delete newState.stacks[action.payload.stackId][action.payload.blockId];
```

Comment:
This is where the bug lies

Reply:
you need to create a new blocks and stacks dictionary. If you see above, when we DATA_SET_BLOCK, we spread the contents of the current state, then spread of the contents of state.blocks. This effectively creates a new blocks dictionary so redux knows to update components that refer to it appropriately. Here, you're only spreading the state, which copies the pre-existing blocks dictionary into the new state. Since you don't override it like it's done in DATA_SET_BLOCK, when you delete an entry from it, you're deleting it from the old dictionary.

try:
newState = {
  ...state,
  blocks: {
     ...state.blocks
  },
  stacks: {
    ...state.stacks,
  }
}

newState.stacks[stackId] = {
  ...state.stacks[stackId],
  order: state.stacks[stackId].order.filter(id => id !== blockId),
}

delete newState.blocks[blockId];

of course, extract blockId and stackId from payload first.
also this might be wrong so try it out and hopefully it makes sense. I'll try to explain later how this works